utils/scripts/helpers/repo-helper-ubuntu.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+set -uex
+
+# This script is used by dockerfiles to optionally use
+# a local repository instead of a distro provided repository.
+# It will also optionally allow running a /tmp/install script
+# for custom packages if present.
+
+: "${REPO_FILE_URL:=}"
+: "${HTTPS_PROXY:=}"
+: "${DAOS_LAB_CA_FILE_UR:=}"
+: "${REPOSITORY_NAME:=artifactory}"
+
+disable_repos () {
+    if [ -e /etc/apt/sources.list.d/ubuntu.sources ];then
+        mv /etc/apt/sources.list.d/ubuntu.sources \
+           etc/apt/sources.list.d/ubuntu.sources.disabled
+    elif [ -e /etc/apt/sources.list ];then
+        mv /etc/apt/sources.list \
+           etc/apt/sources.list.disabled
+    fi
+}
+
+# Use local repo server if present
+install_curl() {
+
+    if command -v curl; then
+        echo "found curl!"
+        return
+    else
+        apt-get update
+        apt-get install curl ca-certificates gpg gpg-agent \
+                software-properties-common
+    fi
+
```

Comment:
Problem #1:
Apparently, the Ubuntu 22.04 docker image has the curl package pre-installed and does not have the rest of the packages so the adding of the ppa:longsleep/golang repository fails.

Shows up from Problem #2:
According to the Ubuntu packaging documentation, only the Ubuntu 20.04 version needs this PPA for golang, and Ubuntu 22.04 and later should not be enabling the repo.